test/test.downscopedclient.ts

```diff
@@ -0,0 +1,406 @@
+            },
+          ],
+        },
+      };
+      assert.doesNotThrow(() => {
+        return new DownscopedClient(client, cabWithTwoAvailblePermissions);
+      });
+    });
+
+    it('should not throw with empty available condition', () => {
+      const cabWithNoAvailabilityCondition = {
+        accessBoundary: {
+          accessBoundaryRules: [
+            {
+              availableResource: testAvailableResource,
+              availablePermissions: [testAvailablePermission1],
+            },
+          ],
+        },
+      };
+      assert.doesNotThrow(() => {
+        return new DownscopedClient(client, cabWithNoAvailabilityCondition);
+      });
+    });
+
+    it('should not throw with only expression setup in available condition', () => {
+      const cabWithOnlyAvailabilityConditionExpression = {
+        accessBoundary: {
+          accessBoundaryRules: [
+            {
+              availableResource: testAvailableResource,
+              availablePermissions: [
+                testAvailablePermission1,
+                testAvailablePermission2,
+              ],
+              availabilityCondition: {
+                expression: testAvailabilityConditionExpression,
+              },
+            },
+          ],
+        },
+      };
+      assert.doesNotThrow(() => {
+        return new DownscopedClient(
+          client,
+          cabWithOnlyAvailabilityConditionExpression
+        );
+      });
+    });
+
+    it('should set custom RefreshOptions', () => {
+      const refreshOptions = {
+        eagerRefreshThresholdMillis: 5000,
+        forceRefreshOnFailure: true,
+      };
+      const cabWithOneAccessBoundaryRules = {
+        accessBoundary: {
+          accessBoundaryRules: [
+            {
+              availableResource: testAvailableResource,
+              availablePermissions: [testAvailablePermission1],
+              availabilityCondition: {
+                expression: testAvailabilityConditionExpression,
+              },
+            },
+          ],
+        },
+      };
+      const downscopedClient = new DownscopedClient(
+        client,
+        cabWithOneAccessBoundaryRules,
+        refreshOptions
+      );
+      assert.strictEqual(
+        downscopedClient.forceRefreshOnFailure,
+        refreshOptions.forceRefreshOnFailure
+      );
+      assert.strictEqual(
+        downscopedClient.eagerRefreshThresholdMillis,
+        refreshOptions.eagerRefreshThresholdMillis
+      );
+    });
+  });
+
+  describe('getAccessToken()', () => {
+    let sandbox: sinon.SinonSandbox;
+    before(() => {
+      const expectedAccessTokenResponse = {
+        token: 'subject_token',
+      };
+      sandbox = sinon.createSandbox();
+      sandbox
+        .stub(client, 'getAccessToken')
+        .resolves(expectedAccessTokenResponse);
+    });
+
+    after(() => {
+      sandbox.restore();
+    });
+
+    it('should return current unexpired cached DownscopedClient access token', async () => {
```

Comment:
Can you also add the following tests:
- Test when STS rejects with an error.
- Test when the source AuthClient rejects when a token is requested.